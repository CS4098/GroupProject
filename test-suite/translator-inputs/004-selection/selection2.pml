process test {
    selection {
        action act1 {
	        requires { a }
	        provides { a }
        }
        task one {
		    action a {
		        requires { a }
		        provides { b }
		    }
		    action b {
		        requires { b }
		        provides { c }
		    }
		    action c {
		        requires { c }
		        provides { d }
		    }
	    }
    }
}

