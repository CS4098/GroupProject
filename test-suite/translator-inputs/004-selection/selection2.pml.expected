bool a;
bool c;
bool b;
bool d;

active proctype test()
{
	if
	:: true ->
			a ->
				a = true;
	:: true ->
				a ->
					b = true;
				b ->
					c = true;
				c ->
					d = true;
	fi
}
